nilearn/_utils/estimator_checks.py

```diff
@@ -882,6 +902,59 @@ def check_img_estimators_overwrite_params(estimator) -> None:
         )
 
 
+@ignore_warnings()
+def check_img_estimators_pickle(estimator_orig):
+    """Test that we can pickle all estimators.
+
+    Adapted from sklearn's check_estimators_pickle
+    """
+    estimator = clone(estimator_orig)
+
+    X, _ = generate_data_to_fit(estimator)
+
+    if isinstance(estimator, NiftiSpheresMasker):
+        # NiftiSpheresMasker needs mask_img to run inverse_transform
+        mask_img = new_img_like(X, np.ones(X.shape[:3]))
+        estimator.mask_img = mask_img
+
+    set_random_state(estimator)
+
+    fitted_estimator = fit_estimator(estimator)
+
+    pickled_estimator = pickle.dumps(fitted_estimator)
+    unpickled_estimator = pickle.loads(pickled_estimator)
+
+    result = {}
+
+    check_methods = ["transform"]
+    input_data = [X] if isinstance(estimator, SearchLight) else [[X]]
+    if hasattr(estimator, "inverse_transform"):
+        check_methods.append("inverse_transform")
+        input_data.append(_rng().random((1, fitted_estimator.n_elements_)))
+
+    for method, input in zip(check_methods, input_data):
+        if hasattr(estimator, method):
+            result[method] = getattr(estimator, method)(input)
+            result["input"] = input
+
+    for method, input in zip(check_methods, input_data):
+        if method not in result:
+            continue
+        unpickled_result = getattr(unpickled_estimator, method)(input)
+        if isinstance(unpickled_result, np.ndarray):
+            if isinstance(estimator, SearchLight):
+                # TODO check why Searchlight has lower absolute tolerance
```

Comment:
To be checked before merging ?

Reply:
Not sure.
Would prefer to circle back it in another PR.